spark_pipeline_framework/utilities/file_downloader.py

```diff
@@ -0,0 +1,137 @@
+import os
+import shutil
+import tempfile
+from typing import Any, Optional
+from urllib import request as url_request
+from urllib import parse as url_parse
+from zipfile import ZipFile, is_zipfile
+
+
+class ThrowOnErrorOpener(url_request.FancyURLopener):
+    def http_error_default(
+        self, url: str, fp: Any, errcode: int, errmsg: str, headers: Any
+    ) -> Any:
+        raise Exception(f"{errcode}: {errmsg}")
+
+
+class FileDownloader:
+    """
+    Module to download files from urls
+    Inspired from python wget module: https://github.com/steveeJ/python-wget
+    """
+
+    def __init__(
+        self, url: str, download_path: Optional[str] = "",
```

Comment:
I think it's likely that the `download_path` be an s3 url

Reply:
Ideally that should be the case, but some of the reference data files are obtained in the form of archive files. So the utility should be able to download the archive files onto the server, extract them and then upload them to the S3.
So the `download_path` in such cases would be a temporary location on the server and s3 url in other cases.

Reply:
I was under the impression that FileDownloader handle unzipping automatically but still, the final destination is very likely s3 which doesn't seem we're handling it in FileDownloader class or anywhere else